phantom_token.c

```diff
@@ -219,17 +218,34 @@ static ngx_int_t handler(ngx_http_request_t *request)
         if (module_context->done)
         {
             // return appropriate status
-            if (module_context->status >= NGX_HTTP_OK && module_context->status < NGX_HTTP_SPECIAL_RESPONSE)
+            if (module_context->status == NGX_HTTP_OK)
             {
                 // Introspection was successful. Replace the incoming Authorization header with one that has the JWT.
                 request->headers_in.authorization->value.len = module_context->jwt.len;
                 request->headers_in.authorization->value.data = module_context->jwt.data;
 
                 return NGX_OK;
             }
+            else if (module_context->status == NGX_HTTP_NO_CONTENT)
+            {
+                return set_www_authenticate_header(request, module_location_config->realm,
+                                                          module_location_config->space_separated_scopes, NULL);;
+            }
+            else if (module_context->status == NGX_HTTP_SERVICE_UNAVAILABLE)
+            {
+                return NGX_HTTP_SERVICE_UNAVAILABLE;
+            }
+            else if (module_context->status >= NGX_HTTP_INTERNAL_SERVER_ERROR)
+            {
+                return NGX_HTTP_BAD_GATEWAY;
+            }
+            else if (module_context->status == NGX_HTTP_UNAUTHORIZED ||
+                     module_context->status == NGX_HTTP_FORBIDDEN || module_context->status == NGX_HTTP_NOT_FOUND)
+            {
+                return NGX_HTTP_BAD_REQUEST;
```

Comment:
if Curity returns 401, 403, or 404 to NGINX, this is more like bad gateway or server error, right?